osquery/tables/networking/freebsd/interface_ip.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <net/if.h>
+#include <netinet/in.h>
+#include <netinet6/in6_var.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/sysctl.h>
+#include <sys/types.h>
+// Small hack to make the header compatible with C++
+#define prf_ra in6_prflags::prf_ra
+#include <netinet6/nd6.h>
+#undef prf_ra
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/networking/posix/interfaces.h>
+
+namespace osquery {
+namespace tables {
+
+namespace {
+const std::unordered_map<std::string, int> kIpv6SysctlObjects = {
+    {"forwarding", IPV6CTL_FORWARDING},
+    {"redirect", IPV6CTL_SENDREDIRECTS},
+    {"hlim", IPV6CTL_DEFHLIM},
+    {"rtadv", IPV6CTL_ACCEPT_RTADV},
```

Comment:
The code for freeBSD and Darwin look pretty similar, would you like to reuse it? To avoid duplication.

Reply:
Yes, the code looks similar because both use `sysctl` to get the config of the IP stack.
I took a look at `osquery/tables/networking/{freebsd,darwin}/routes.cpp` and the problem is the same: the code is very similar.
I don't really know how to refactor the code: I don't want to put the functions in `posix/` because Linux doesn't implements these sysctl interfaces (only Darwin and BSD are sharing the code).

Reply:
Probably you could wrap up some common parts of the code to the separated functions. What do you think?